### PR TITLE
restack v3

### PR DESCRIPTION
**Pull request by ahdinosaur**
:tophat: :cat:

- `dogstack`
  - [x] split api and asset server
  - [x] remove lint, dev, and test cli commands
  - [x] change dogstack from magical framework into importable library
  - [x] ~make one `server` cli command (server can be exported `createApiServer()` or `createAssetServer()`)~
    - maybe remove `babel-register` from cli command
  - [x] update README.md documentation
- `dogstack-example`
  - [x] get example working
  - [x] simply browser config into single `browser.js`
  - [x] ~use `dog server api.js` and `dog server asset.js`~
  - [x] deploy to heroku and netlify
  - [x] update documentation
- `create-dogstack-app`
  - [x] update app generator to generate initial app

### CODE REVIEW

**Comment by iainkirkpatrick** [comment]
been down a path of attempting to solve the (apparent) problem of 'multiple copies of React' by upgrading `material-ui` - which initially seemed to fix that problem, but eventually blocked by `redux-form-material-ui` not yet fully supporting v1 of `material-ui` https://github.com/erikras/redux-form-material-ui/pull/249/files

File: README.md [hunk 1]
```diff
@@ -178,26 +174,28 @@ export default {
...
 
+example:
 ```js
 // client.js
 export default {
-  services: []
+  services: [
+    authentication
+  ],
+  apiUrl: config.api.url
```
Comment (ahdinosaur) [inline README.md]: given that we prescribe the shape of `config.api.url`, would it be more friendly to just take in the whole config?

File: README.md [hunk 2]
```diff
@@ -125,43 +107,57 @@ const hooks = {
 }
 ```
 
-### `store.js`
+### `browser.js`
+
+a function to generate your dogstack client app, which takes as an arg an object of the following properties:
```
Comment (ahdinosaur) [inline README.md]: i think we should include an example `browser.js` importing from the various files described below. i was kinda confused why each the files were described when it's just to use in `browser.js`.

File: createLog.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ function createLog (options = {}) {
   const {
     name,
     level = process.env.LEVEL || 'info',
```
Comment (ahdinosaur) [inline createLog.js]: this should use `config.log.level` which in production takes `process.env.LOG_LEVEL`

File: transform.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const configify = require('./lib/configify')
+module.exports = function transform (filename, options) {
```
Comment (ahdinosaur) [inline transform.js:2]: this is missing from the README, your app wouldn't work without it!

**Review by ahdinosaur — COMMENTED** [review]
some bits and bops

**Comment by ahdinosaur** [comment]
@iainkirkpatrick here's some things in the peachy bins: https://github.com/peachcloud/peachhub-provider/blob/master/api/bin.js

File: README.md (hunk 1 above)
Comment (iainkirkpatrick) [reply README.md]: 👍 as discussed

File: README.md (hunk 2 above)
Comment (iainkirkpatrick) [reply README.md]: 👍

File: createLog.js (hunk 3 above)
Comment (iainkirkpatrick) [reply createLog.js]: moved calls + code around so `createLog` could know about `config` 👍

File: transform.js (hunk 4 above)
Comment (iainkirkpatrick) [reply transform.js:2]: 😅 added a piece for it 🙏

**Pull request merged**